### PR TITLE
enhancement(console sink, codecs): Integrate `encoding::Encoder` with `console` sink

### PR DESCRIPTION
**Pull request by pablosichert**
Part of #9459.

### CODE REVIEW

File: src/sinks/console/sink.rs [hunk 1]
```diff
@@ -27,40 +42,55 @@ where
...
     async fn run(mut self: Box<Self>, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
-        while let Some(event) = input.next().await {
+        if matches!(
+            (self.encoder.serializer(), self.encoder.framer()),
+            (
+                codecs::encoding::Serializer::Json(_),
+                Framer::CharacterDelimited(CharacterDelimitedEncoder { delimiter: b',' }),
+            )
+        ) {
+            self.write_all(b"[").await?;
```
Comment (lukesteensen) [inline src/sinks/console/sink.rs:52]: This is a little odd. Are we sure it's worthwhile to support JSON vs just newline-delimited JSON for a streaming sink like this?

Comment (pablosichert) [reply src/sinks/console/sink.rs]: > This is a little odd. Are we sure it's worthwhile to support JSON vs just newline-delimited JSON for a streaming sink like this?

I agree there. However, we currently don't explicitly disallow users from configuring the framing method to `character_delimited`, so they could still configure that if they really wanted

Comment (pablosichert) [reply src/sinks/console/sink.rs]: > Yeah, agreed, this sink should only use ndjson not json arrays.

For what it's worth, even when setting the `json` `StandardsEncoding`, this sink also uses `newline_delimited` framing in this PR

File: lib/codecs/src/encoding/format/raw_message.rs [hunk 2]
```diff
@@ -48,7 +48,7 @@ impl Encoder<Event> for RawMessageSerializer {
                 .get_by_meaning(message_key)
                 .or_else(|| log.get(message_key)) // backward compatibility
                 .map(|value| value.coerce_to_bytes()),
-            Event::Metric(_) => None,
+            Event::Metric(metric) => Some(metric.to_string().into()),
```
Comment (lukesteensen) [inline lib/codecs/src/encoding/format/raw_message.rs:51]: Do we want to support metrics in `RawMessageSerializer`? This seems at odds with [the sink input type PR](https://github.com/vectordotdev/vector/pull/12561/files#diff-49ff9ead8250b219380497e0df68eea7c79e8c2e8af9b3bcffe42507f5eb7153=).

Comment (jszwedko) [reply lib/codecs/src/encoding/format/raw_message.rs:51]: Yeah, thinking about this more. I think there are two serializers here: `binary` and `text`. For `binary` only logs should be supported. `text` should support encoding metrics for use with the `console` sink (and others). I think OpenMetrics, as we do now, is a good option.

Comment (pablosichert) [reply lib/codecs/src/encoding/format/raw_message.rs:51]: This has been resolved with #12754, going to update this PR accordingly

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**